contracts/base/BaseVault.sol

```diff
@@ -4,49 +4,48 @@ pragma solidity ^0.8.9;
 
 import { ERC20PresetMinterPauser as CollateralToken } from '@openzeppelin/contracts/token/ERC20/presets/ERC20PresetMinterPauser.sol';
 import { OwnableUpgradeable } from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
-import { IERC20 } from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
 import { IERC20Metadata } from '@openzeppelin/contracts/interfaces/IERC20Metadata.sol';
 
 import { IClearingHouse } from '@ragetrade/core/contracts/interfaces/IClearingHouse.sol';
-import { IClearingHouseStructures } from '@ragetrade/core/contracts/interfaces/clearinghouse/IClearingHouseStructures.sol';
-import { IVToken } from '@ragetrade/core/contracts/interfaces/IVToken.sol';
 import { SignedMath } from '@ragetrade/core/contracts/libraries/SignedMath.sol';
 import { SignedFullMath } from '@ragetrade/core/contracts/libraries/SignedFullMath.sol';
 import { AddressHelper } from '@ragetrade/core/contracts/libraries/AddressHelper.sol';
 import { UniswapV3PoolHelper, IUniswapV3Pool } from '@ragetrade/core/contracts/libraries/UniswapV3PoolHelper.sol';
 import { Extsload } from '@ragetrade/core/contracts/utils/Extsload.sol';
 import { ClearingHouseExtsload } from '@ragetrade/core/contracts/extsloads/ClearingHouseExtsload.sol';
 
-import { ERC20 } from '@rari-capital/solmate/src/tokens/ERC20.sol';
-
 import { FullMath } from '@uniswap/v3-core-0.8-support/contracts/libraries/FullMath.sol';
 import { FixedPoint96 } from '@uniswap/v3-core-0.8-support/contracts/libraries/FixedPoint96.sol';
 
 import { IBaseVault } from '../interfaces/IBaseVault.sol';
+import { ISwapSimulator } from '../interfaces/ISwapSimulator.sol';
 import { IBaseYieldStrategy } from '../interfaces/IBaseYieldStrategy.sol';
 
+import { ClearingHouseLens } from '@ragetrade/core/contracts/lens/ClearingHouseLens.sol';
+
 import { RageERC4626 } from './RageERC4626.sol';
 import { SafeCast } from '../libraries/SafeCast.sol';
 import { Logic } from '../libraries/Logic.sol';
 
 abstract contract BaseVault is IBaseVault, RageERC4626, IBaseYieldStrategy, OwnableUpgradeable, Extsload {
     using AddressHelper for address;
-    using AddressHelper for IVToken;
 
-    using SafeCast for int256;
     using SafeCast for uint256;
     using SignedMath for int256;
     using SignedFullMath for int256;
-    using UniswapV3PoolHelper for IUniswapV3Pool;
     using ClearingHouseExtsload for IClearingHouse;
```

Comment:
I had a skim through the solidity files, just wanted to post a note here that ClearingHouseLens contract + ClearingHouseExtsload library is used. ClearingHouseExtsload is cheaper but adds bytecode to the vault contract while ClearingHouseLens is just an extra CALL over the gas cost and low on bytecode. So in future if we are short on bytecode, we can remove ClearingHouseExtsload and use ClearingHouseLens instead.